tools/ctl/linux/aws-otel-collector-ctl

```diff
@@ -44,20 +44,41 @@ UsageString="
   status:                                 get the status of the agent process.
 
   -c: configuration
-  file:<file-path>:                       file path on the host
+  <config-uri>:                           - file path on the host. E.g.: /tmp/config.yaml
+                                          - http uri. E.g.: http://example.com/config
+                                          - https uri. E.g.: https://example.com/config
+                                          - s3 uri. E.g.: s3://bucket/config
 
   "
 
-aoc_start() {
+aoc_config_remote_uri() {
     config="${1:-}"
 
+    echo "config=\"--config ${config}\"" > $ENV_FILE
+}
+
+aoc_config_local_uri() {
+    config="${1:-}"
+
+    echo "config=\"--config /opt/aws/aws-otel-collector/etc/config.yaml\"" > $ENV_FILE
+
     if [ ! -z $config ] && [ -f $config ]; then
         cp $config $CONFDIR/config.yaml
     fi
 
     if [ ! -f $CONFDIR/config.yaml ]; then
         cp $DFT_CONFDIR/.config.yaml $CONFDIR/config.yaml
     fi
+}
+
+aoc_start() {
+    config="${1:-}"
+
+    if [[ $config =~ ^(https|http|s3)://.* ]]; then
```

Comment:
This seems brittle.  We'll have to change this regex every time a new config provider is added.

Reply:
I will replace by a more relaxed regex.